### PR TITLE
API for Multi-Level Pie Chart is added

### PR DESCRIPTION
**Pull request by himanshu-balasamanta**

### CODE REVIEW

**Comment by eselmeister** [comment]
@Himanshu-Balasamanta The class "CompressMultiLevelPie" is missing.

File: org.eclipse.swtchart/src/org/eclipse/swtchart/internal/series/MultiLevelPie.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	@Override
+	public Range getAdjustedRange(Axis axis, int length) {
+
+		return null;
+	}
+
+	@Override
+	protected void setCompressor() {
+
+		((CompressMultiLevelPie)compressor).update();
```
Comment (Himanshu-Balasamanta) [inline org.eclipse.swtchart/src/org/eclipse/swtchart/internal/series/MultiLevelPie.java:123]: Kindly comment out this line and run the "MultiLevelPieChartExample" class to run without errors

**Comment by himanshu-balasamanta** [comment]
Hi.
I was reluctant to add it now as it becomes a bit more difficult to review the code. But to run the example smoothly, it was necessary to add the class. Now if this is fine with the reviewers, I shall update the legend to support multi-level pie chart in my next PR.

**Comment by eselmeister** [comment]
@Himanshu-Balasamanta Your code looks good. I have one question and one issue I'd like you to have a look at:

1) CompressMultiLevelPie
The members are package protected. Is this on purpose?
```
	IdNodeDataModel model;
	HashMap<String, Node> data;
	List<Map<Integer, Node>> levelNodes;
```

2) PieChartExample
The MultiLevelPieExample runs fine, but when I try to run the PieChartExample, the following error is thrown:

```
Exception in thread "main" java.lang.NullPointerException
	at org.eclipse.swtchart.internal.Legend.paintControl(Legend.java:441)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:231)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5686)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1370)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1396)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1379)
	at org.eclipse.swt.widgets.Control.gtk_expose_event(Control.java:3663)
	at org.eclipse.swt.widgets.Composite.gtk_expose_event(Composite.java:812)
	at org.eclipse.swt.widgets.Widget.windowProc(Widget.java:1979)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:6468)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5904)
	at org.eclipse.swt.internal.gtk.GTK._gtk_main_do_event(Native Method)
	at org.eclipse.swt.internal.gtk.GTK.gtk_main_do_event(GTK.java:4084)
	at org.eclipse.swt.widgets.Display.eventProc(Display.java:1383)
	at org.eclipse.swt.internal.gtk.OS._g_main_context_iteration(Native Method)
	at org.eclipse.swt.internal.gtk.OS.g_main_context_iteration(OS.java:1626)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4516)
	at org.eclipse.swtchart.examples.PieChartExample.main(PieChartExample.java:49)
```

**Comment by himanshu-balasamanta** [comment]
Kindly try now.

**Pull request merged**